infermedica_api/models/rationale.py

```diff
@@ -0,0 +1,18 @@
+from .base import BaseModel
+
+
+class RationaleResult(BaseModel):
+    """Model class for handling rationale result object."""
+
+    @staticmethod
+    def from_json(json):
+        """
+        Constructs RationaleResult object from given dict and returns it.
+
+        :param json: Dict with a rationale result value.
+        :type json: dict
+
+        :returns: RationaleResult object
+        :rtype: :class:`infermedica_api.models.RationaleResult`
+        """
+        return RationaleResult(**json)
```

Comment:
If we are doing models here, we could have also add a model for `observation_params` fields returned by rationale endpoint. Which is a list of objects (potentially RationaleResultObservationParam) containing id, name and common_name. Similarly `condition_params`.

Reply:
To be honest, I think we should generally define a structure of our models, whereas currently we have some models being just a wrappers on top of a json dicts and some having a few hints as to what will they contain.

Reply:
I agree, as we discussed, we should consider removing the models. Defining the exact models can be tricky in terms of changes in the API it self (this library will then require much frequent updates). If it is not a problem let's implement it as is for consistency reasons and we can come back to subject of version 2.0 of this lib at any time, we will have some time to refactor that.

Reply:
Do you mean to keep it as it is or should I add these additional fields?

Reply:
Let's leave it ;)